elementary/monitor/dbt_project/macros/cte/current_tests_run_results.sql

```diff
@@ -0,0 +1,248 @@
+{% macro current_tests_run_results_query(days_back = none) %}
+    {% set columns = adapter.get_columns_in_relation(ref('elementary', 'elementary_test_results')) %}
+    {% set columns_names = [] %}
+    {% for column in columns %}
+        {% do columns_names.append(column.column | lower) %}
+    {% endfor %}
+    {% if 'test_short_name' not in columns_names or 'test_alias' not in columns_names %}
+        with elementary_test_results as (
+            select * from {{ ref('elementary_test_results') }}
+            {% if days_back %}
+                where {{ elementary.datediff(elementary.cast_as_timestamp('detected_at'), elementary.current_timestamp(), 'day') }} < {{ days_back }}
+            {% endif %}
+        ),
+        
+        dbt_tests as (
+            select * from {{ ref('dbt_tests') }}
+        ),
+        
+        first_time_test_occurred as (
+            select 
+                min(detected_at) as first_time_occurred,
+                test_unique_id
+            from elementary_test_results
+            group by test_unique_id
+        )
+        
+        select
```

Comment:
I tend to disagree here.
IMO it is really good to be explicit with the column you use and pass.
Using * (or a loop that automatically generates the column names) might make problems as much as forgetting to add a column (braking schemas, breaking unions, group by and etc).

When you develop a new feature you need to test it anyhow and make sure that everything works well (and we have tests for that + it is part of the release process), so it will be raised if the new column is needed 🙂

Reply:
Yeah I guess you are right, it makes sense for queries.
Mainly thinking about it from a duplication perspective but it's better not to break schemas.
Sababa :)